### DISCUSSION TITLE
In QSelect multiple option is not working as expected

### DISCUSSION THREAD

**Discussion by sanket-parte** [elimintz/justpy#697] [General]
Hello everyone,

We are currently utilizing the QSelect widget within a form, specifically with the multiple=true argument enabled. However, we've encountered an issue where, upon selecting multiple options in the QSelect dropdown, the widget does not transmit all the selected option IDs to the backend upon form submission. Strangely, it only seems to send a single ID (the ID of the first selected option) in string format. Consequently, we're unable to successfully store and save multiple selected options in our database.

Here's the QSelect widget definition for reference:
![select_role_def](https://github.com/justpy-org/justpy/assets/43114999/c57423c3-118d-4755-aa7b-548a066fb6f9)

And here's the corresponding form definition:
![form](https://github.com/justpy-org/justpy/assets/43114999/2f44cf1c-8d65-4e0f-ac6e-6965d6823e34)

For clarity, I've attached an image illustrating the widget after selecting multiple options:
![select_role_widget](https://github.com/justpy-org/justpy/assets/43114999/82674820-b069-4b09-9dde-00cfa3472343)

If anyone has encountered a similar issue or has insights into how to ensure that all selected option IDs are properly transmitted to the backend upon form submission, we would greatly appreciate your assistance.

Thank you in advance for your time and expertise.

**Comment by harshalizode** [comment]
Hello, 
We also faced a similar issue with the QSelect, It may be justpy form unable to parse the list, that is why in form_data we getting the single ID as a string. 

To solve this issue I take the values from the form components list. 
In our callback function, we have the form instance,  in that we get the components list that we used in our form along with its values. 
You can also use that until get a proper approach to deal with it.

**Comment by freemant2000** [comment]
This is not a justpy bug; it is how QSelect works. It is not really a html <select>. Internally it consists many different element but inside there is only one &lt;input&gt; which is used to allow the user to perform keyword search and to allow the html label to work:
![image](https://github.com/justpy-org/justpy/assets/1174319/1e2d94ca-c20f-4eda-9f9a-2badef4bae8f)

On the other hand, with Justpy the component's value is updated instantly, so there is no need to wait until form submission. You can also access the "value" attribute of your QSelect to get the latest value. It works when multiple selection is enabled too.

**Reply by harshalizode** [reply]
Ok, got it.
Thanks for your reply.
